crawlerr/services/strategies/repository_strategy.py

```
@@ -0,0 +1,398 @@
+                raise Exception("Clone directory was not created")
+
+            # Remove .git directory to save space
+            git_dir = target_dir / ".git"
+            if git_dir.exists():
+                shutil.rmtree(git_dir)
+
+            self.logger.info(f"Successfully cloned repository to {target_dir}")
+            return target_dir
+
+        except TimeoutError:
+            raise Exception("Repository cloning timed out (5 minutes)")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)